js/search.js

```diff
@@ -87,52 +90,39 @@
 
 		action: function () {
 			var val = this.$input.val();
-			var inputEmptyOrUnchanged = (val === '' || val === this.activeSearch);
-
-			if (this.activeSearch && inputEmptyOrUnchanged) {
-				this.clear();
-			} else if (val) {
+			if (val && val.length > 0) {
 				this.search(val);
 			}
+			else {
+				this.clear();
+			}
 		},
 
 		buttonclicked: function (e) {
 			e.preventDefault();
 			if ($(e.currentTarget).is('.disabled, :disabled')) return;
-			this.action();
-		},
 
-		keypress: function (e) {
-			if (e.which === 13) {
-				e.preventDefault();
+			if(this.$element.hasClass('searched')) {
+				this.clear();
+			}
+			else {
+				this.action();
 			}
 		},
 
-		keypressed: function (e) {
-			var remove = 'glyphicon-remove';
-			var search = 'glyphicon-search';
-			var val;
-
+		keypress: function (e) {
 			if (e.which === 13) {
+				// enter pressed
```

Comment:
Nice to have:
instead of the `Unchanged` part of  `inputEmptyOrUnchanged` add a `escape` event listener for `e.which === 27`

Also add comments above each number or create a constant so code is easier to read
```
var ENTER_KEY_CODE = 13;
var TAB_KEY_CODE = 9;
var ESC_KEY_CODE = 27;
```